### PR TITLE
add edit functionality for editing the fixed parts of the rtp header

### PR DESCRIPTION
**Pull request by ystreet**

### CODE REVIEW

File: src/edit.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+impl From<RtpWriteError> for RtpPacketEditError {
+    fn from(value: RtpWriteError) -> Self {
+        Self::Write(value)
+    }
+}
+
+/// Packet editing
+pub struct RtpPacketEdit<'a> {
+    data: &'a mut [u8],
```
Comment (sdroege) [inline src/edit.rs]: If you make this and `RtpPacket` `#[repr(transparent)]` then you could `Deref<Target=RtpPacket>` via a simple transmute and get direct access to all the getters

File: src/edit.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+impl From<RtpWriteError> for RtpPacketEditError {
+    fn from(value: RtpWriteError) -> Self {
+        Self::Write(value)
+    }
+}
+
+/// Packet editing
+pub struct RtpPacketEdit<'a> {
```
Comment (sdroege) [inline src/edit.rs]:
```suggestion
pub struct RtpPacketMut<'a> {
```

?

Comment (ystreet) [reply src/edit.rs]: `Mut` is probably a little better for modify in place.  I do wonder what name should be given to a potential modify with size change structure.  `OwnedMut`? `MutOwned`?  `OwnedMut` may not be necessary though and am overthinking this a little bit.

**Review by sdroege — APPROVED** [review]

File: src/edit.rs (hunk 2 above)
Comment (sdroege) [reply src/edit.rs]: I'd say we worry about that once we need it :)

**Pull request merged**